sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapRfRExamples.cs

```diff
@@ -0,0 +1,184 @@
+                        upsertComplexMarketDataRequest.Add("discount_curve_CHF",
+                            TestDataUtilities.BuildRateCurveRequest(TestDataUtilities.EffectiveAt, "CHF", "OIS",
+                                TestDataUtilities.ExampleDiscountFactors1));
+                        upsertComplexMarketDataRequest.Add("projection_curve",
+                            TestDataUtilities.BuildRateCurveRequest(TestDataUtilities.EffectiveAt, "CHF", "SARON",
+                                TestDataUtilities.ExampleDiscountFactors2, "1D"));
+                        break;
+                }
+                upsertComplexMarketDataResponse =
+                    _complexMarketDataApi.UpsertComplexMarketData(scope, upsertComplexMarketDataRequest);
+                ValidateComplexMarketDataUpsert(upsertComplexMarketDataResponse,
+                    upsertComplexMarketDataRequest.Count);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void GetAndValidatePortfolioCashFlows(
+            LusidInstrument instrument,
+            string scope, string portfolioCode,
+            string recipeCode,
+            string instrumentID)
+        {
+            var swap = (InterestRateSwap) instrument;
+            var cashflows = _transactionPortfoliosApi.GetPortfolioCashFlows(
+                scope: scope,
+                code: portfolioCode,
+                effectiveAt: TestDataUtilities.EffectiveAt.AddYears(3),
+                windowStart: swap.StartDate.AddDays(-3),
+                windowEnd: swap.MaturityDate.AddDays(3),
+                asAt:null,
+                filter:null,
+                recipeIdScope: scope,
+                recipeIdCode: recipeCode).Values;
+            
+            Assert.That(cashflows.Count, Is.GreaterThanOrEqualTo(1));
+        }
+
+        [LusidFeature("F5-23")]
+        [TestCase("SOFR")]
+        [TestCase("ESTR")]
+        [TestCase("SONIA")]
+        [TestCase("TONA")]
+        [TestCase("SARON")]
+        public void InterestRateSwapRfRCreationAndUpsertionExample(string type)
+        {
+            // CREATE an interest rate swap (that can then be upserted into LUSID)
+            var swap = InstrumentExamples.CreateExampleInterestRateSwap(type);
+            
+            // ASSERT that it was created
+            Assert.That(swap, Is.Not.Null);
+
+            // CAN NOW UPSERT TO LUSID
+            var uniqueId = swap.InstrumentType + Guid.NewGuid().ToString(); 
+            var instrumentsIds = new List<(LusidInstrument, string)>{(swap, uniqueId)};
+            var definitions = TestDataUtilities.BuildInstrumentUpsertRequest(instrumentsIds);
+            
+            var upsertResponse = _instrumentsApi.UpsertInstruments(definitions);
+            ValidateUpsertInstrumentResponse(upsertResponse);
+
+            // CAN NOW QUERY FROM LUSID
+            var getResponse = _instrumentsApi.GetInstruments("ClientInternal", new List<string> { uniqueId });
```

Comment:
you may need to qualify the `GetInstruments` with the `asAt` returned from `UpsertInstruments` do avoid any DEI eventual consistency issues

Reply:
There shouldn't be an issue as the uniqueID is genrated during this the test and only one version is upserted. The asat is optional:
<param name="asAt">The asAt datetime at which to retrieve the instrument definitions. Defaults to returning the latest version of each instrument definition if not specified. (optional)

Can't see there being any conflicts when this test is run - even if run in parallel?

Reply:
its less about the uniqueness of the id and more to do with whether the instrument will be returned from the list call. It _may_ not get returned if it hasn't been indexed, but specifying the `asAt` on the list will guarantee it is returned

Reply:
Ok - will need to change all the tests that use this.